vispy/scene/cameras/cameras.py

```diff
@@ -151,60 +428,61 @@ def _set_scene_transform(self, tr):
         else:
             self._transform_cache.roll()
             self._scene_transform = self._transform_cache.get([pre_tr, tr])
-            
-        if self.viewbox is not None:
-            self.viewbox.scene.transform = self._scene_transform
-            self.viewbox.update()
+        
+        # Update scene    
+        self._viewbox.scene.transform = self._scene_transform
+        self._viewbox.update()
+        
+        # Apply same state to linked cameras, but prevent that camera
+        # to return the favor
+        if not self._linked_cameras_no_update:
+            for cam in self._linked_cameras:
+                cam._linked_cameras_no_update = True
+                try:
+                    cam.set_state(self.get_state())
+                finally:
+                    cam._linked_cameras_no_update = False
 
     
 class PanZoomCamera(BaseCamera):
```

Comment:
My impression looking at this class is that its implementation was much simpler when it just used `self._rect` to keep track of its state, and the added complexity is a result of trying to force it to use scale_factor, center, and aspect ratio defined in BaseCamera. Are there new features here that I'm not accounting for?